### PR TITLE
Use sqlite for caching

### PR DESCRIPTION
**Pull request by jackrobison**

### CODE REVIEW

File: lighthouse/updater/MetadataUpdater.py [hunk 1]
```diff
@@ -13,167 +15,319 @@
...
+    parsed_lines = [l.split('=') for l in lines]
+    d_for_return = {k: v for k, v in parsed_lines}
+    return d_for_return
+
+
+def save_settings(path, settings):
+    f = open(path, "w")
+    lines = ["%s=%s" % (k, v) for k, v in settings.iteritems()]
+    f.writelines(lines)
+    f.close()
```
Comment (jobevers) [inline lighthouse/updater/MetadataUpdater.py]: Why are we defining our own settings format?  Can we use yml or some other standard library?

File: lighthouse/search/search.py [hunk 2]
```diff
@@ -134,9 +134,29 @@ def _format(results):
...
 
+        def _direct_match(search, results):
+            results_for_return = results
+            if search in self.updater.metadata:
+                # if the name isn't in the results, add it as the top result
+                if search not in [r[0] for r in results]:
+                    results_for_return.pop()
+                    results_for_return.reverse()
+                    results_for_return.append((search, 1000))
+                    results_for_return.reverse
```
Comment (jobevers) [inline lighthouse/search/search.py]: rewrite this as something like:

```
if not check_results(search, results):
    add_to_top(results_for_return, search)
```

File: lighthouse/search/search.py [hunk 3]
```diff
@@ -134,9 +136,29 @@ def _format(results):
...
+
+                # if the name is in the results but in the wrong position, move it to the front
+                wrong_position = next((r for r in results[1:] if r[0] == search), False)
+                if wrong_position:
+                    results_for_return.remove(wrong_position)
+                    results_for_return.reverse()
+                    results_for_return.append(wrong_position)
+                    results_for_return.reverse()
```
Comment (jobevers) [inline lighthouse/search/search.py]: same here:

```
wrong_position = some_descriptive_fn_name(results, search)
if wrong_position:
    move_to_top(results_for_return, wrong_position)
```

File: lighthouse/search/search.py (hunk 2 above)
Comment (jobevers) [inline lighthouse/search/search.py]: Can then unit test `check_results` and `add_to_top`

**Comment by lyoshenka** [comment]
Why is db/lighthouse.sqlite included?

**Comment by lyoshenka** [comment]
Also, can you add tests to this please? Even the most basic tests. We need to start validating that things actually work as expected.

**Pull request merged**